### PR TITLE
Mac initialization task 1

### PR DESCRIPTION
**Pull request by boyuan12**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs change / refactoring / dependency upgrade

## Motivation and Context / Related issue

Added MAC initialization, no longer hard-coded to 0.

fixed issue #5 

## How Has This Been Tested (if it applies)
Added tests according to the issue description

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: test/test_mpc.py [hunk 1]
```diff
@@ -59,10 +59,8 @@ def _check(self, encrypted_tensor, reference, msg, dst=None, tolerance=None):
             logging.info("Result - Reference = %s" % (tensor - reference))
         self.assertTrue(test_passed, msg=msg)
         self.assertTrue(hasattr(encrypted_tensor, "_mac"), "_mac attribute failed to propagate")
-        self.assertEqual(encrypted_tensor._mac, 0, "_mac attribute has incorrect value")
-
-        self.assertTrue(hasattr(encrypted_tensor, "_mac"), "_mac attribute failed to propagate")
-        self.assertEqual(encrypted_tensor._mac, 0, "_mac attribute has incorrect value")
+        if dst is None:
+           self.assertTrue(encrypted_tensor._tensor.reveal().mul(MPCTensor.alpha.reveal()).eq(encrypted_tensor._mac.reveal()).all(), msg="Mac Arithmetic Mismatch")
```
Comment (knottb) [inline test/test_mpc.py]: This is a good start. I think there is a little more we should do before closing this.

Here, we are only testing that the mac is correct when `dst is None`. In order to test this properly, we need to test both when `dst is None` and when `dst` takes a value.

So, we should add some `elif` blocks here for the following:

Case 1: Current process is not the `dst` party. Here we want the result to be None.

Case 2: Current process is the `dst` party. Here we want the test to be the same as above.



```
elif dst != self.rank:
    self.assertIsNone(encrypted_tensor._mac.reveal(dst=dst)
else:
    # same as above code.
```

In order to avoid copy-pasting code, these can be consolidated:

```
if dst is None or dst == self.rank:
    # Above code with reveal(dst=dst) for each reveal
elif dst is not None:
    self.assertIsNone(encrypted_tensor._mac.reveal(dst=dst))
```

Let me know if you have questions.

**Review by knottb — CHANGES_REQUESTED** [review]

File: test/test_mpc.py (hunk 1 above)
Comment (boyuan12) [reply test/test_mpc.py]: One quick question, what exactly  `.rank` attribute do?

Comment (boyuan12) [reply test/test_mpc.py]: Also, after I modified test to this:
```
if dst is None or dst == self.rank:
    self.assertTrue(encrypted_tensor._tensor.reveal().mul(MPCTensor.alpha.reveal()).eq(encrypted_tensor._mac.reveal()).all(), msg="Mac Arithmetic Mismatch")
elif dst is not None:
    self.assertIsNone(encrypted_tensor._mac.reveal(dst=dst))
```

It hanged again on `self.assertTrue(encrypted_tensor._tensor.reveal().mul(MPCTensor.alpha.reveal()).eq(encrypted_tensor._mac.reveal()).all(), msg="Mac Arithmetic Mismatch")` for some reason.

Comment (smsunarto) [reply test/test_mpc.py]: It seems to be stuck only when dst == self.rank
